reco_utils/evaluation/python_evaluation.py

```diff
@@ -94,31 +102,26 @@ def merge_rating_true_pred(
         col_prediction (str): column name for prediction.
 
     Returns:
-        pd.DataFrame: Merged pd.DataFrame
-    """
-    # Select the columns needed for evaluations
-    rating_true = rating_true[[col_user, col_item, col_rating]]
-    rating_pred = rating_pred[[col_user, col_item, col_prediction]]
+        np.array: Array with the true ratings
+        np.array: Array with the predicted ratings
 
-    if col_rating == col_prediction:
-        rating_true_pred = pd.merge(
+    """
+    suffixes = ["_true", "_pred"]
+    # Apart from merging both dataframes, pd.merge will rename the columns with the suffixes only if the rating
+    # column name of rating_true is the same as the name rating column name in rating_pred
+    rating_true_pred = pd.merge(
             rating_true,
             rating_pred,
             on=[col_user, col_item],
-            suffixes=["_true", "_pred"],
-        )
-        rating_true_pred.rename(
-            columns={col_rating + "_true": DEFAULT_RATING_COL}, inplace=True
-        )
-        rating_true_pred.rename(
-            columns={col_prediction + "_pred": PREDICTION_COL}, inplace=True
+            suffixes=suffixes,
         )
+    if col_rating == col_prediction:
+        column_select_true = col_rating + suffixes[0]
+        column_select_pred = col_prediction + suffixes[1]
     else:
-        rating_true_pred = pd.merge(rating_true, rating_pred, on=[col_user, col_item])
-        rating_true_pred.rename(columns={col_rating: DEFAULT_RATING_COL}, inplace=True)
-        rating_true_pred.rename(columns={col_prediction: PREDICTION_COL}, inplace=True)
-
-    return rating_true_pred
+        column_select_true = col_rating
+        column_select_pred = col_prediction
+    return rating_true_pred[column_select_true], column_select_true[column_select_pred]
```

Comment:
typo on second variable there - should be rating_true_pred[column_select_pred]

Reply:
thanks, fixed now